### PR TITLE
[NCL-5230] Show ErrorResponse message in client exceptions

### PR DESCRIPTION
**Pull request by janinko**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: rest-api/src/main/java/org/jboss/pnc/client/RemoteResourceException.java [hunk 1]
```diff
@@ -17,33 +17,45 @@
...
 /**
  * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
  */
 public class RemoteResourceException extends ClientException {
 
+    @Getter
     private final int status;
 
+    @Getter
+    private final ErrorResponse response;
```
Comment (matejonnet) [inline rest-api/src/main/java/org/jboss/pnc/client/RemoteResourceException.java]: What about making it Optional, to protect users form NPE?

File: rest-api/src/main/java/org/jboss/pnc/client/RemoteResourceException.java [hunk 2]
```diff
@@ -17,33 +17,45 @@
...
-        status = -1;
+        this.status = -1;
+        this.response = null;
     }
 
     public RemoteResourceException(ClientErrorException cause) {
         super(cause);
-        Response response = cause.getResponse();
-        status = response.getStatus();
+        this.status = cause.getResponse().getStatus();
```
Comment (matejonnet) [inline rest-api/src/main/java/org/jboss/pnc/client/RemoteResourceException.java]: is this NPE prof ? Is getResponse() always not null ?

Comment (janinko) [reply rest-api/src/main/java/org/jboss/pnc/client/RemoteResourceException.java]: I just put it on one line, I didn't change behaviour :)

But it should be always set AFAIK.

File: processor/src/main/java/org/jboss/pnc/processor/ClientGenerator.java [hunk 3]
```diff
@@ -206,7 +206,6 @@ private boolean process0(Set<? extends TypeElement> annotations, RoundEnvironmen
                         MethodSpec.Builder methodBuilder = beginMethod(restApiMethod);
                         addDefaultParameters(restApiMethod, methodBuilder);
                         methodBuilder
-                                .addException(ClassName.get("org.jboss.pnc.client", "ClientException"))
```
Comment (matejonnet) [inline processor/src/main/java/org/jboss/pnc/processor/ClientGenerator.java]: Why did you remove the exception from the message signature ?

Comment (janinko) [reply processor/src/main/java/org/jboss/pnc/processor/ClientGenerator.java]: Because this exception is not thrown anywhere.

**Comment by janinko** [comment]
retest this please
